backends/ent/backend/store.go

```diff
@@ -0,0 +1,246 @@
+// --------------------------------------------------------------
+// SPDX-FileCopyrightText: Copyright © 2024 The Protobom Authors
+// SPDX-FileType: SOURCE
+// SPDX-License-Identifier: Apache-2.0
+// --------------------------------------------------------------
+package backend
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"slices"
+
+	"github.com/bom-squad/protobom/pkg/sbom"
+	sqlite "github.com/glebarez/go-sqlite"
+
+	"github.com/protobom/storage/backends/ent"
+	"github.com/protobom/storage/backends/ent/documenttype"
+	ent_node "github.com/protobom/storage/backends/ent/node"
+	"github.com/protobom/storage/pkg/options"
+)
+
+// Enable SQLite foreign key support.
+const dsnParams string = "?_pragma=foreign_keys(1)"
+
+type (
+	// EntBackend implements the protobom model.v1.storage.Backend interface.
+	EntBackend struct{}
+
+	// EntOptions contains options specific to the protobom ent backend.
+	EntOptions struct {
+		DatabaseFile string
+	}
+)
+
+var errInvalidEntOptions = errors.New("invalid ent backend options")
+
+// NewEntOptions returns an EntOptions with optional file path for database file.
+//
+// If not provided, the database file will default to ":memory:" for a temporary, in-memory database.
+func NewEntOptions(file string) EntOptions {
+	if file == "" {
+		file = ":memory:"
+	}
+
+	return EntOptions{DatabaseFile: file}
+}
```

Comment:
As there will be more options than just the filename, I suggest not taking the path here as the function will quickly be rendered obsolete. I would rather create a default set and return it:

```suggestion
var defaultEntOptions = EntOptions {
   DatabaseFile: ":memory:",
}
func NewEntOptions() EntOptions {
	return defaultEntOptions
}
```

If you want to define the options at creation time you could use the functional options pattern, we use it in the main protobom project:

https://github.com/protobom/protobom/blob/a5791ce702610e88fa2be32ee3ad41aa2c5c1d5d/pkg/writer/options.go#L10-L18

Reply:
I made a first pass at this. I wanted to make it flexible, so I tried implementing it in such a way that a developer can write either

```go
// Functional options pattern
ent.NewBackend(ent.WithDatabaseFile("sqlite.db"))
```

or

```go
// Builder pattern
ent.NewBackend().WithDatabaseFile("sqlite.db")
```

If you don't like having both options I can remove one.

Reply:
SGTM :+1: